pkgs/os-specific/linux/power-profiles-daemon/default.nix

```diff
@@ -43,26 +68,101 @@ stdenv.mkDerivation rec {
     libxml2 # for xmllint for stripping GResources
     libxslt # for xsltproc for building docs
     gobject-introspection
+    wrapGAppsNoGuiHook
+    python3.pkgs.wrapPython
+
+    # For finding tests.
+    (python3.withPackages testPythonPkgs)
   ];
 
   buildInputs = [
     libgudev
     systemd
     upower
     glib
-    (python3.withPackages (ps: with ps; [ ps.pygobject3 ])) # for cli tool
+    polkit
+    python3 # for cli tool
+  ];
+
+  strictDeps = true;
+
+  # for cli tool
+  pythonPath = [
+    python3.pkgs.pygobject3
   ];
 
   mesonFlags = [
+    "-Dinstalled_test_prefix=${placeholder "installedTests"}"
     "-Dsystemdsystemunitdir=${placeholder "out"}/lib/systemd/system"
     "-Dgtk_doc=true"
   ];
 
+  PKG_CONFIG_POLKIT_GOBJECT_1_POLICYDIR = "${placeholder "out"}/share/polkit-1/actions";
+
+  # Avoid double wrapping
+  dontWrapGApps = true;
+
+  postPatch = ''
+    patchShebangs tests/unittest_inspector.py
+  '';
+
+  preConfigure = ''
+    # For finding tests.
+    GI_TYPELIB_PATH_original=$GI_TYPELIB_PATH
+    addToSearchPath GI_TYPELIB_PATH "${testTypelibPath}"
+    GI_TYPELIB_PATH_original=$GI_TYPELIB_PATH
+  '';
+
+  postConfigure = ''
+    GI_TYPELIB_PATH=$GI_TYPELIB_PATH_original
+    unset GI_TYPELIB_PATH_original
+  '';
```

Comment:
What is this for? It doesn't make a difference in the output.

Reply:
Thanks, fixed.

Reply:
It looked like it was supposed to stop extra dependncies from getting in the output?

Reply:
Oh that is what it does, just export was missing. I wasn't sure why it was not working.

Reply:
The issue was actually that I was overwriting GI_TYPELIB_PATH_original after changing it.